### PR TITLE
Newsletter: Remove Subscribe Modal from Settings -> Discussion

### PR DESCRIPTION
**Pull request by lezama**
Removes duplicate setting; the setting is available at Settings -> Newsletter.

## Proposed changes:

* Before:
Screenshot 2024-08-09 at 10 52 06 AM

* After:
Screenshot 2024-08-09 at 10 45 53 AM

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Testing instructions:

* Go to Settings -> Discussions
* Check that saving works.

### CODE REVIEW

File: projects/plugins/jetpack/modules/subscriptions.php [hunk 1]
```diff
@@ -369,21 +369,6 @@ public function configure() {
...
-
-		add_settings_field(
-			'jetpack_subscriptions_comment_subscribe',
-			__( 'Enable Subscribe Modal', 'jetpack' ),
-			array( $this, 'subscribe_modal_setting' ),
-			'discussion',
-			'jetpack_subscriptions'
-		);
-
-		register_setting(
```
Comment (simison) [inline projects/plugins/jetpack/modules/subscriptions.php:382]: Does this affect anything remaining, i.e. still possible to toggle the setting at Newsletter settings page?

**Review by simison — APPROVED** [review]
Pending a [question](https://github.com/Automattic/jetpack/pull/38805/files#r1711602609), looks good.

The duplicate setting is gone from wp-admin.

File: projects/plugins/jetpack/modules/subscriptions.php (hunk 1 above)
Comment (lezama) [reply projects/plugins/jetpack/modules/subscriptions.php:382]: > still possible to toggle the setting at Newsletter settings page

yes, confirmed 😄

**Comment by simison** [comment]
cc @StefMattana this change might need some updates around docs? 👋

**Comment by StefMattana** [comment]
@jos22 or @foleynotrose I don't see any mention of these WP Admin settings in the external documentation. 
I checked [here](https://jetpack.com/support/jetpack-blocks/subscription-form-block/) and [here](https://jetpack.com/support/newsletter/), and no mentions; I also found nothing else on other pages. Should we add it somewhere? My gut is that it's a minor change in the WP Admin on a setting that's very down to the Discussion page, and we don't seem to refer users to this page in our docs. 

Correct me if I'm wrong!

**Comment by simison** [comment]
We'd like to remove them as duplicates so better not add yet. :-)

**Comment by StefMattana** [comment]
I think this is a good idea as a duplicate settings may get users confused rather than helping them :)

**Pull request merged**